lib/pure/strutils.nim

```diff
@@ -2860,6 +2860,79 @@ proc isNilOrWhitespace*(s: string): bool {.noSideEffect, procvar, rtl,
   ## Alias for isEmptyOrWhitespace
   result = isEmptyOrWhitespace(s)
 
+
+since (1, 1):
+  template toLowerAsciiInPlace(c: var char) =
+    if c in {'A'..'Z'}: c = chr(ord(c) + (ord('a') - ord('A')))
+
+  template toUpperAsciiInPlace(c: var char) =
+    if c in {'a'..'z'}: c = chr(ord(c) - (ord('a') - ord('A')))
```

Comment:
Please add `{.inline.}` to the current `toLowerAscii` and `toUpperAscii` instead of duplicating the code here. It's demonstrated that the performance gains of this duplication is [negligible compared to when you add `{.inline.}` to the current versions](https://github.com/nim-lang/Nim/pull/13610#discussion_r390030324).

Reply:
Done.